adal/src/telemetry/java/com/microsoft/aad/adal/BrokerEvent.java

```diff
@@ -57,12 +61,26 @@ void setBrokerAccountServiceConnected() {
         setProperty(EventStrings.BROKER_ACCOUNT_SERVICE_CONNECTED, Boolean.toString(true));
     }
 
+    void setBrokerAccountServiceConnectionErrorInfo(final String errorInfo) {
+        setProperty(EventStrings.BROKER_ACCOUNT_SERVICE_CONNECTION_ERROR_INFO, errorInfo);
+    }
+
+    void setBrokerAccountServiceConnectionErrorInfo(final Throwable throwable) {
+        if (sEnableStackTraces && null != throwable) {
+            setProperty(EventStrings.BROKER_ACCOUNT_SERVICE_CONNECTION_ERROR_INFO, Log.getStackTraceString(throwable));
```

Comment:
What will happen if GDPR is disabled? Are we still logging the stack trace into the telemetry?

Reply:
The stack trace will be sent regardless - shouldn't contain any PII/OII (see outputs above, no object state is leaked)